hack/tas-kind-cluster.yaml

```diff
@@ -0,0 +1,155 @@
+---
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+  - role: control-plane
+    kubeadmConfigPatches:
+      - |
+        kind: ClusterConfiguration
+        apiVersion: kubeadm.k8s.io/v1beta3
+        scheduler:
+          extraArgs:
+            v: "2"
+        controllerManager:
+          extraArgs:
+            v: "2"
+        apiServer:
+          extraArgs:
+            enable-aggregator-routing: "true"
+            v: "2"
+  - role: worker
+    labels:
+      instance-type: on-demand
+      cloud.provider.com/topology-block: b1
+      cloud.provider.com/topology-rack: r1
+    kubeadmConfigPatches:
+      - |
+        kind: InitConfiguration
+        nodeRegistration:
+          kubeletExtraArgs:
+            v: "2"
+      - |
+        kind: JoinConfiguration
+        nodeRegistration:
+          kubeletExtraArgs:
+            cpu-manager-policy: static
+            kube-reserved: "cpu=8,memory=1Gi" # Reserves CPU and memory for Kubernetes system to reduce Allocatable resources
```

Comment:
This looks a bit tricky. Maybe for the first iteration we can use virtual resources as susggested in https://github.com/kubernetes-sigs/kueue/pull/3489/files/85b53b86d0a4e59f15c23044cff82a4ddf35c885#r1840209137. 
I see testing cpu and memory could be important, but the reality is that most users will use it for GPUs, and thus it seems the more interesting case actually.

Reply:
that's actually really good idea, if I can only add the extra resources then it will work

Reply:
you can probably add them in the BeforeSuite

Reply:
the setup is anyway a bit artificial as the kind nodes aren't really split into racks :) 
But this is something we also were cheating about in the past - the idea of "on-demand" flavor is for nodes which are provisioned "on-demand" by CA, but don't exist at the moment of cluster creation :)